File: official/landing_pages/built_for_developers/create_shipment.rb
# Create To and From Addresses
from_address = EasyPost::Address.create(...)
to_address = EasyPost::Address.create(...)

# Create Shipment and Get Rates
shipment = EasyPost::Shipment.create(
  from_address: from_address,
  to_address: to_address,
  parcel: {
    predefined_package: 'Parcel',
    weight: 32.8
  }
)

# Buy Shipping Label
shipment.buy(rate: shipment.lowest_rate())

File: official/landing_pages/built_for_developers/create_shipment.py
import os

import easypost


client = easypost.EasyPostClient(os.getenv("EASYPOST_API_KEY"))

shipment = client.shipment.create(
    to_address={"id": "adr_..."},
    from_address={"id": "adr_..."},
    parcel={
        "predefined_package": "Parcel",
        "weight": 32.8,
    },
)

client.shipment.buy(
    shipment.id,
    rate=shipment.lowest_rate(),
)
